java/main/src/main/java/com/saucelabs/saucebindings/SauceSession.java

```diff
@@ -118,6 +121,121 @@ public void stop(Boolean passed) {
+        driver.executeScript("sauce: disable log");
+    }
+
+    /**
+     * Turns logging back on after logging has been disabled
+     * @see #disableLogging()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void enableLogging() {
+        validateSessionStarted("enableLogging()");
+        driver.executeScript("sauce: enable log");
+    }
+
+    /**
+     * Stops VM’s network connection.
+     * @see #startNetwork()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void stopNetwork() {
+        validateSessionStarted("stopNetwork()");
+        SaucePlatform platformName = sauceOptions.getPlatformName();
+        if (!platformName.isMac()) {
+            throw new InvalidArgumentException("Can only stop network for a Mac Platform; current platform is: " + platformName);
+        }
+
+        driver.executeScript("sauce: stop network");
+    }
+
+    /**
+     * Restarts the VM’s network connection after it has been stopped
+     * @see #stopNetwork()
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     */
+    public void startNetwork() {
+        validateSessionStarted("startNetwork()");
+        SaucePlatform platformName = sauceOptions.getPlatformName();
+        if (!platformName.isMac()) {
+            throw new InvalidArgumentException("Can only stop network for a Mac Platform; current platform is: " + platformName);
+        }
+
+        driver.executeScript("sauce: start network");
+    }
+
+    /**
+     * This is typically set in SauceOptions before starting the Session.
+     * It might be useful to set dynamically in some situations (e.g., making failing tests more visible
+     *
+     * TODO: Implement this with API because this is not currently working
+     *
+     * @param visibility who can see the job details in the Sauce Labs UI
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     * @see BaseConfigurations#setJobVisibility(JobVisibility)
+     */
+    public void setVisibility(JobVisibility visibility) {
+        validateSessionStarted("setJobVisibility()");
+        driver.executeScript("sauce:job-info={'public': '" + visibility.getValue() + "'");
+    }
+
+    /**
+     * This is typically set in SauceOptions before starting the Session.
+     * This allows dynamially updating if necessary (e.g. name based on discovered state).
+     *
+     * @param name the test name
+     * @see <a href="https://docs.saucelabs.com/basics/test-config-annotation/test-annotation/#methods">Test Annotation Methods</a>
+     * @see BaseConfigurations#setName(String)
+     */
+    public void changeName(String name) {
```

Comment:
Same here, `changeTestName()`?

Reply:
oops forgot to change this last time. just did.